gear/gear/metrics.py

```diff
@@ -25,3 +28,38 @@ async def monitor_endpoints_middleware(request, handler):
         raise e
     finally:
         CONCURRENT_REQUESTS.labels(endpoint=endpoint, verb=verb).dec()
+
+
+class LoggingTimer:
+    def __init__(self, query_name, threshold_ms=None):
+        self.query_name = query_name
+        self.threshold_ms = threshold_ms
+        self.timing = {}
+        self.start_time = None
+        self.sql_query_latency_manager = None
+
+    def step(self, name):
+        return LoggingTimerStep(self, name)
+
+    async def __aenter__(self):
+        SQL_QUERY_COUNT.labels(query_name=self.query_name).inc()
```

Comment:
This is fine for now, but let's follow up later with another PR that gets us closer to matching the information that we have for HTTP endpoints. Specifically, I would want to examine the `exc` in `__aexit__` and add a sort of `status` label to these metrics. It would be nice to know, for example, if a query is slow only when it is failing or encountering deadlocks, but is otherwise fast.